src/apps/ipv6/reassemble.lua

```diff
@@ -264,24 +284,27 @@ function Reassembler:handle_fragment(h)
       else
          reassembly.final_start = frag_start
       end
-   elseif frag_size % 8 ~= 0 then
+   elseif bit.band(frag_size, 0x7) ~= 0 then
```

Comment:
This changed isn't justified. I benchmarked % and "mod with logical and" and they're the same speed. What's indeed much slower is `math.fmod`. I think this change should be reverted.

Reply:
Ok, I assume that `%` optimizes for this case?

Reply:
The results I got from benchmarking % and & showed no difference. Here's a gist of the code I tried https://gist.github.com/dpino/925a79494d3da27e3681e8b3acf67b32. I don't know what LuaJIT compares % to, though.

Reply:
I just checked and the compiler does indeed specialize to bitops if the module is a power of 2.